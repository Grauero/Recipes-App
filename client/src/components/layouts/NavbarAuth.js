import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import Signout from '../auth/Signout';

const NavbarAuth = ({ session }) => (
  <Fragment>
    <ul>
      <li>
        <NavLink exact to="/">
          Home
        </NavLink>
      </li>
      <li>
        <NavLink to="/search">Search</NavLink>
      </li>
      <li>
        <NavLink to="/recipe/add">Add Recipe</NavLink>
      </li>
      <li>
        <NavLink to="/profile">Profile</NavLink>
      </li>
      <li>
        <Signout />
      </li>
    </ul>
    <h4>
      Welcome, <strong>{session.getCurrentUser.username}</strong>
    </h4>
  </Fragment>
);

NavbarAuth.propTypes = {
  session: PropTypes.shape({
    getCurrentUser: PropTypes.shape({
      username: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default NavbarAuth;
