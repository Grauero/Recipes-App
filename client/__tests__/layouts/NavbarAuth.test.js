import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import NavbarAuth from '../../src/components/layouts/NavbarAuth';

const props = { session: { getCurrentUser: { username: 'username' } } };
let component;

beforeEach(() => {
  component = shallow(<NavbarAuth {...props} />);
});

afterEach(() => {
  component.unmount();
});

it('matches snapshot', () => {
  expect(toJSON(component)).toMatchSnapshot();
});

it('renders username', () => {
  const heading = component.find('h4');

  expect(heading.text()).toBe(`Welcome, ${props.session.getCurrentUser.username}`);
});
