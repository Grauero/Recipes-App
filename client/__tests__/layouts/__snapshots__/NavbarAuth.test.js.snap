// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <ul>
    <li>
      <NavLink
        activeClassName="active"
        aria-current="page"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/search"
      >
        Search
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/recipe/add"
      >
        Add Recipe
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/profile"
      >
        Profile
      </NavLink>
    </li>
    <li>
      <withRouter(Signout) />
    </li>
  </ul>
  <h4>
    Welcome, 
    <strong>
      username
    </strong>
  </h4>
</Fragment>
`;
